src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.wasm.cs

```diff
@@ -112,12 +112,14 @@ private protected override void OnIsEnabledChanged(IsEnabledChangedEventArgs e)
 			ApplyEnabled(e.NewValue);
 		}
 
-		partial void OnIsReadonlyChangedPartial(DependencyPropertyChangedEventArgs e)
+		partial void OnIsReadonlyChangedPartial(DependencyPropertyChangedEventArgs e) => UpdateTextBoxViewIsReadOnly();
+
+		partial void OnIsTabStopChangedPartial() => UpdateTextBoxViewIsReadOnly();
+
+		private void UpdateTextBoxViewIsReadOnly()
 		{
-			if (e.NewValue is bool isReadonly)
-			{
-				ApplyIsReadonly(isReadonly);
-			}
+			var isReadOnly = IsReadOnly || !IsTabStop;
```

Comment:
Does not having tabstop really means that the textbox is readonly? This feels like the naming should be different for the function.

Reply:
The difference here should be that when `IsReadOnly = true`, the control should be read-only, but the managed one (e.g. the `TextBox` itself) should still be focusable. In case of `IsTabStop = false` it is not focusable anymore. In terms of the behavior of the `TextBoxView` however (in this case `<input />`) - in both cases we want to make it `readonly` in HTML. So I think the naming in this case should be ok

Reply:
maybe `isNativeReadOnly`/`ApplyNativeReadOnly`?

and you should also put the explanation from your reply as a source comment ;)

Reply:
Realized the `ApplyIsReadonly` method itself is unnecessary indirection, so moved the code directly to the `UpdateTextBoxViewReadonly` method and adjusted the naming